### PR TITLE
Add TinyGo Serverless AI documentation

### PR DESCRIPTION
**Pull request by tpmccallum**
Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title, `template`, and `date` are all set
- [x] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [x] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)

✅ content/spin/go-components.md
tpmccallum@192-168-1-15 developer % bart check content/spin/language-support-overview.md 
✅ content/spin/language-support-overview.md
tpmccallum@192-168-1-15 developer % bart check content/spin/serverless-ai-api-guide.md  
✅ content/spin/serverless-ai-api-guide.md
tpmccallum@192-168-1-15 developer % bart check content/spin/ai-sentiment-analysis-api-tutorial.md 
✅ content/spin/ai-sentiment-analysis-api-tutorial.md

- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)

Screenshot 2023-10-11 at 21 17 13

- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [x] Have tested with `npm run test` and resolved all errors

Screenshot 2023-10-11 at 21 17 50

- [x] Relates to an existing (potentially outdated) blog article? If so please add URL in blog to point to this content.

### CODE REVIEW

File: content/spin/ai-sentiment-analysis-api-tutorial.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ In this tutorial we will:
 
 * Update Spin (and dependencies) on your local machine
 * Create a Serverless AI application
-* Learn about the Serverless AI SDK (in Rust, TypeScript and Python)
+* Learn about the Serverless AI SDK (in Rust, TypeScript, Python and TinyGo)
```
Comment (itowlson) [inline content/spin/ai-sentiment-analysis-api-tutorial.md:42]: We can probably drop this list now it's "all our current languages"

File: content/spin/ai-sentiment-analysis-api-tutorial.md [hunk 2]
```diff
@@ -137,6 +147,22 @@ HTTP path: /api/...
...
+
+> Note: please add `/api/...` when prompted for the path; this provides us with an API endpoint to query the sentiment analysis component.
+
+<!-- @selectiveCpy -->
+
+```bash
+$ spin new http-go 
+Enter a name for your new application: sentiment-analysis
+Description: A sentiment analysis API that demonstrates using LLM inferencing and KV stores together
+HTTP base: /
```
Comment (itowlson) [inline content/spin/ai-sentiment-analysis-api-tutorial.md:160]: This is no longer prompted for in the v2 templates

**Review by itowlson — APPROVED** [review]
Couple of things we might want to update for v2 but we can pick those up in later work

File: content/spin/ai-sentiment-analysis-api-tutorial.md (hunk 2 above)
Comment (tpmccallum) [reply content/spin/ai-sentiment-analysis-api-tutorial.md]: And done.

**Pull request merged**